src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs

```diff
@@ -195,7 +195,15 @@ public override long Speed
             try
             {
                 string path = Path.Combine(NetworkFiles.SysClassNetFolder, name, NetworkFiles.SpeedFileName);
-                return StringParsingHelpers.ParseRawLongFile(path);
+                long megabitsPerSecond = StringParsingHelpers.ParseRawLongFile(path);
+                if (megabitsPerSecond == -1)
+                {
+                    return megabitsPerSecond;
+                }
+                else
+                {
+                    return megabitsPerSecond * 1_000_000; // Value must be returned in bits per second, not megabits.
+                }
```

Comment:
@danmosemsft It's to avoid returning -1,000,000, actually. If the value is -1, then return that, otherwise multiply by 1,000,000.